llama-index-integrations/tools/llama-index-tools-oracleai/llama_index/tools/oracleai/base.py

```diff
@@ -0,0 +1,202 @@
+# -----------------------------------------------------------------------------
+# Authors:
+#   Harichandan Roy (hroy)
+#   David Jiang (ddjiang)
+#
+# -----------------------------------------------------------------------------
+# ...tools/oracleai.py
+# -----------------------------------------------------------------------------
+
+from __future__ import annotations
+
+import json
+import logging
+import traceback
+from typing import Any, Dict, List, Optional, TYPE_CHECKING
+
+from llama_index.core.schema import Document
+
+if TYPE_CHECKING:
+    from oracledb import Connection
+
+logger = logging.getLogger(__name__)
+
+
+"""OracleSummary class"""
+
+
+class OracleSummary:
+    """Get Summary.
```

Comment:
This is a tool integration, but doesn't follow the same base class as every other tool? How do I use this with an agent?

Reply:
We did not find any tool related to summary. So we put it as an independent class. Users familiar with Oracle APIs can use this class to serve their usecases.

Reply:
There won't be any tools specific to summary, rather, there is a general base class that you can extend. Maybe there is a misunderstanding on the term `tools` ? This is referring to tools that LLMs can invoke to help answer user questions or accomplish tasks.

Reading this class closer, it seems more like a generic utils function rather than something an agent would use. Maybe this should be under `llama-index-utils` ?

For examples of actual tools, see this for example:
https://github.com/run-llama/llama_index/blob/main/llama-index-integrations/tools/llama-index-tools-arxiv/llama_index/tools/arxiv/base.py

The doctstring becomes the tool description, and the tool can be used in any function calling agent/llm

Reply:
Ok hari will make this change.